### PR TITLE
bugfix/collectible detection by acccount

### PR DESCRIPTION
**Pull request by estebanmino**
When auto detecting collectibles there are some cases where some collectibles were detected in the wrong account. One way I could reproduce that was in the following case

- Auto detection is triggered for address `0x9` and we're waiting for the response from API https://github.com/MetaMask/controllers/compare/bugfix/assets-detection-by-acccount?expand=1#diff-3202c2273084eb58dae3e0d4e23765dce262bb24c93eb3bf166c56d2d7bf02c8R365
- The account is changed to `0x12`
- The API response resolves the collectibles for `0x9` but they are added to `0x12`, so we have the wrong assets into the account.

I could repro that with tests, removing this check https://github.com/MetaMask/controllers/compare/bugfix/assets-detection-by-acccount?expand=1#diff-3202c2273084eb58dae3e0d4e23765dce262bb24c93eb3bf166c56d2d7bf02c8R401

### CODE REVIEW

File: src/assets/AssetsDetectionController.ts [hunk 1]
```diff
@@ -355,9 +355,10 @@ export class AssetsDetectionController extends BaseController<
     if (!this.isMainnet()) {
       return;
     }
-    const { selectedAddress } = this.config;
+    const requestedSelectedAddress = this.config.selectedAddress;
```
Comment (rickycodes) [inline src/assets/AssetsDetectionController.ts:358]: any particular reason for the rename? seems `selectedAddress` would still be fine no?

**Review by rickycodes — APPROVED** [review]
one small thing, but otherwise looks good to me.

File: src/assets/AssetsDetectionController.ts (hunk 1 above)
Comment (estebanmino) [reply src/assets/AssetsDetectionController.ts:358]: hm not really, just to be clear that it was the address from where it was being called

Comment (estebanmino) [reply src/assets/AssetsDetectionController.ts:358]: I can go back to selectedAddress if you want 👌

Comment (rickycodes) [reply src/assets/AssetsDetectionController.ts:358]: nah it's fine.

> just to be clear that it was the address from where it was being called

I think that's plenty reason enough to rename. feel free to merge.

**Pull request merged**